File: test/mothers/blog/post.mother.dart
import 'package:staretz/blog/domain/entities/post.dart';
import 'package:staretz/blog/domain/value_objects/post_body.dart';
import 'package:staretz/blog/domain/value_objects/post_excerpt.dart';
import 'package:staretz/blog/domain/value_objects/post_id.dart';
import 'package:staretz/blog/domain/value_objects/post_image_url.dart';
import 'package:staretz/blog/domain/value_objects/post_published_at.dart';
import 'package:staretz/blog/domain/value_objects/post_slug.dart';
import 'package:staretz/blog/domain/value_objects/post_title.dart';

class PostMother {
  static Post valid() => Post.create(
        id: PostId.create('1'),
        title: PostTitle.create('Test Post'),
        slug: PostSlug.create('test-post'),
        imageUrl: PostImageUrl.create('https://example.com/image.jpg'),
        excerpt: PostExcerpt.create('A short excerpt.'),
        body: PostBody.create('Full body content.'),
        publishedAt: PostPublishedAt.create(DateTime(2026, 1, 1)),
      );

  static Post withSlug(String slug) => Post.create(
        id: PostId.create('2'),
        title: PostTitle.create('Post $slug'),
        slug: PostSlug.create(slug),
        imageUrl: PostImageUrl.create('https://example.com/image.jpg'),
        excerpt: PostExcerpt.create('Excerpt for $slug.'),
        body: PostBody.create('Body for $slug.'),
        publishedAt: PostPublishedAt.create(DateTime(2026, 6, 1)),
      );
}

File: src/modules/blog/domain/entities/__tests__/Post.mother.ts
import { Post } from '../Post.entity'
import { Category } from '../../value-objects/Category.valueObject'
import { PostAuthor } from '../../value-objects/PostAuthor.valueObject'
import { PostContent } from '../../value-objects/PostContent.valueObject'
import { PostExcerpt } from '../../value-objects/PostExcerpt.valueObject'
import { PostImage } from '../../value-objects/PostImage.valueObject'
import { PostTitle } from '../../value-objects/PostTitle.valueObject'
import { PublishedAt } from '../../value-objects/PublishedAt.valueObject'
import { Slug } from '../../value-objects/Slug.valueObject'

export class PostMother {
  static random(): Post {
    return Post.create({
      slug: Slug.create('sample-post'),
      title: PostTitle.create('Sample Post'),
      excerpt: PostExcerpt.create('A sample excerpt used for tests.'),
      content: PostContent.create('Sample post body content.'),
      author: PostAuthor.create('Jane Doe'),
      publishedAt: PublishedAt.create(new Date('2026-01-01T00:00:00Z')),
      category: Category.create('Architecture'),
      image: PostImage.create('https://picsum.photos/seed/sample-post/1200/800'),
    })
  }

  static empty(): Post {
    return Post.empty()
  }

  static withSlug(slug: string): Post {
    return Post.create({
      slug: Slug.create(slug),
      title: PostTitle.create('Sample Post'),
      excerpt: PostExcerpt.create('A sample excerpt used for tests.'),
      content: PostContent.create('Sample post body content.'),
      author: PostAuthor.create('Jane Doe'),
      publishedAt: PublishedAt.create(new Date('2026-01-01T00:00:00Z')),
      category: Category.create('Architecture'),
      image: PostImage.create(`https://picsum.photos/seed/${slug}/1200/800`),
    })
  }

  static publishedAt(date: Date): Post {
    return Post.create({
      slug: Slug.create('sample-post'),
      title: PostTitle.create('Sample Post'),
      excerpt: PostExcerpt.create('A sample excerpt used for tests.'),
      content: PostContent.create('Sample post body content.'),
      author: PostAuthor.create('Jane Doe'),
      publishedAt: PublishedAt.create(date),
      category: Category.create('Architecture'),
      image: PostImage.create('https://picsum.photos/seed/sample-post/1200/800'),
    })
  }

  static category(category: string): Post {
    return Post.create({
      slug: Slug.create('sample-post'),
      title: PostTitle.create('Sample Post'),
      excerpt: PostExcerpt.create('A sample excerpt used for tests.'),
      content: PostContent.create('Sample post body content.'),
      author: PostAuthor.create('Jane Doe'),
      publishedAt: PublishedAt.create(new Date('2026-01-01T00:00:00Z')),
      category: Category.create(category),
      image: PostImage.create('https://picsum.photos/seed/sample-post/1200/800'),
    })
  }

  static titled(title: string): Post {
    return Post.create({
      slug: Slug.create('sample-post'),
      title: PostTitle.create(title),
      excerpt: PostExcerpt.create('A sample excerpt used for tests.'),
      content: PostContent.create('Sample post body content.'),
      author: PostAuthor.create('Jane Doe'),
      publishedAt: PublishedAt.create(new Date('2026-01-01T00:00:00Z')),
      category: Category.create('Architecture'),
      image: PostImage.create('https://picsum.photos/seed/sample-post/1200/800'),
    })
  }
}
